File: product-service/src/main/java/com/intexsoft/webshop/productservice/dto/product/ProductDto.java
package com.intexsoft.webshop.productservice.dto.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.intexsoft.webshop.productservice.dto.attributevalue.AttributeValueDto;
import com.intexsoft.webshop.productservice.dto.image.ImageDto;
import com.intexsoft.webshop.productservice.dto.subcategory.SubcategoryDto;
import com.intexsoft.webshop.productservice.dto.vendor.VendorDto;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ProductDto {
    Long id;
    String name;
    @JsonProperty("subcategory")
    SubcategoryDto subcategoryDto;
    @JsonProperty("vendor")
    VendorDto vendorDto;
    @JsonProperty("images")
    List<ImageDto> imageDtos;
    @JsonProperty("attributeValues")
    List<AttributeValueDto> attributeValueDtos;
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/dto/product/ProductDto.kt
package com.intexsoft.webshop.productservicekt.dto.product

import com.fasterxml.jackson.annotation.JsonProperty
import com.intexsoft.webshop.productservicekt.dto.attributevalue.AttributeValueDto
import com.intexsoft.webshop.productservicekt.dto.image.ImageDto
import com.intexsoft.webshop.productservicekt.dto.subcategory.SubcategoryDto
import com.intexsoft.webshop.productservicekt.dto.vendor.VendorDto

data class ProductDto(
    val id: Long,
    val name: String,
    val orderQuantity: Short = 0,
    @JsonProperty("subcategory")
    val subcategoryDto: SubcategoryDto?,
    @JsonProperty("vendor")
    val vendorDto: VendorDto?,
    @JsonProperty("images")
    val imageDtos: List<ImageDto>?,
    @JsonProperty("attributeValues")
    val attributeValueDtos: List<AttributeValueDto>?
)